### ISSUE TITLE
现已无法直接按步骤安装环境

### ISSUE THREAD

**Issue by TheLolita** [thuiar/TEXTOIR#55]
![image](https://github.com/thuiar/TEXTOIR/assets/59789293/86dda8cf-ce7b-4453-a77c-6b92dcd76370)
![image](https://github.com/thuiar/TEXTOIR/assets/59789293/d6dc5cfa-4941-4987-836d-18998a639c17)

**Comment by dkgee** [comment]
按说明创建的python3.6环境，pip install -r requirements.txt 一安装就包各种错。真是无语……

**Comment by dkgee** [comment]
按说明创建的python3.6环境，pip install -r requirements.txt 一安装就包各种错。真是无语……

**Comment by giter159** [comment]
> 按说明创建的新的虚拟的python3.6环境，pip install -r requirements.txt 一安装就包各种错。真是无语……

you may need to add some package according to the prompt

**Comment by dkgee** [comment]
其他什么都正常，就是这个requirements.txt里面依赖和python3.6环境不兼容，各种报错啊
mkl-fft>=1.2.0          # 版本不对   python3.8却能正常安装
mkl-random==1.2.2       # 版本不对
mkl-service==2.4.0      # 找不到
sklearn==0.0           # 报错

**Comment by dkgee** [comment]
其他什么都正常，就是这个requirements.txt里面依赖和python3.6环境不兼容，各种报错啊
mkl-fft>=1.2.0          # 版本不对   python3.8却能正常安装
mkl-random==1.2.2       # 版本不对
mkl-service==2.4.0      # 找不到
sklearn==0.0           # 报错

**Comment by giter159** [comment]
> 其他什么都正常，就是这个requirements.txt里面依赖和python3.6环境不兼容，各种报错啊 mkl-fft>=1.2.0 # 版本不对 python3.8却能正常安装 mkl-random==1.2.0 # 版本不对 mkl-service==2.3.0 # 找不到 sklearn==0.0 # 报错

absl-py                      2.1.0
aiohttp                      3.9.3
aiosignal                    1.3.1
astunparse                   1.6.3
async-timeout                4.0.3
attrs                        23.2.0
blessed                      1.20.0
blinker                      1.7.0
boto3                        1.34.72
botocore                     1.34.72
Brotli                       1.1.0
cached-property              1.5.2
cachetools                   5.3.3
certifi                      2024.2.2
cffi                         1.16.0
charset-normalizer           3.3.2
click                        8.1.7
contourpy                    1.1.1
cryptography                 42.0.5
cycler                       0.12.1
easydict                     1.12
filelock                     3.13.1
flatbuffers                  23.5.26
fonttools                    4.49.0
frozenlist                   1.4.1
fsspec                       2024.2.0
gast                         0.4.0
google-auth                  2.28.1
google-auth-oauthlib         0.4.6
google-pasta                 0.2.0
gpustat                      1.1.1
grpcio                       1.54.3
h5py                         3.10.0
huggingface-hub              0.21.2
idna                         3.6
importlib-metadata           7.0.1
importlib_resources          6.1.2
Jinja2                       3.1.3
jmespath                     1.0.1
joblib                       1.3.2
keras                        2.11.0
kiwisolver                   1.4.5
libclang                     16.0.6
Markdown                     3.5.2
MarkupSafe                   2.1.5
matplotlib                   3.7.5
mpmath                       1.3.0
multidict                    6.0.5
networkx                     3.1
nltk                         3.8.1
numpy                        1.24.4
nvidia-cublas-cu12           12.1.3.1
nvidia-cuda-cupti-cu12       12.1.105
nvidia-cuda-nvrtc-cu12       12.1.105
nvidia-cuda-runtime-cu12     12.1.105
nvidia-cudnn-cu12            8.9.2.26
nvidia-cufft-cu12            11.0.2.54
nvidia-curand-cu12           10.3.2.106
nvidia-cusolver-cu12         11.4.5.107
nvidia-cusparse-cu12         12.1.0.106
nvidia-ml-py                 12.535.133
nvidia-nccl-cu12             2.19.3
nvidia-nvjitlink-cu12        12.3.101
nvidia-nvtx-cu12             12.1.105
oauthlib                     3.2.2
opt-einsum                   3.3.0
packaging                    23.2
pandas                       2.0.3
pillow                       10.2.0
pip                          24.0
protobuf                     3.19.6
psutil                       5.9.8
pyasn1                       0.5.1
pyasn1-modules               0.3.0
pycparser                    2.21
PyJWT                        2.8.0
pyOpenSSL                    24.0.0
pyparsing                    3.1.1
PySocks                      1.7.1
python-dateutil              2.8.2
pytorch-pretrained-bert      0.6.2
pytz                         2024.1
pyu2f                        0.1.5
PyYAML                       6.0.1
regex                        2023.12.25
requests                     2.31.0
requests-oauthlib            1.3.1
rsa                          4.9
s3transfer                   0.10.1
safetensors                  0.4.2
scikit-learn                 1.3.2
scipy                        1.10.1
sentence-transformers        2.4.0
setuptools                   69.1.1
six                          1.16.0
sympy                        1.12
tensorboard                  2.11.2
tensorboard-data-server      0.6.1
tensorboard-plugin-wit       1.8.1
tensorflow                   2.13.1
tensorflow-estimator         2.11.0
tensorflow-io-gcs-filesystem 0.34.0
termcolor                    2.4.0
threadpoolctl                3.3.0
tokenizers                   0.15.2
torch                        2.2.1
tqdm                         4.66.2
transformers                 4.38.1
triton                       2.2.0
typing_extensions            4.10.0
tzdata                       2024.1
urllib3                      1.26.18
wcwidth                      0.2.13
Werkzeug                     3.0.1
wheel                        0.42.0
wrapt                        1.16.0
yarl                         1.9.4
zipp                         3.17.0         参考下这些包

**Comment by dkgee** [comment]
> > 其他什么都正常，就是这个requirements.txt里面依赖和python3.6环境不兼容，各种报错啊 mkl-fft>=1.2.0 # 版本不对 python3.8却能正常安装 mkl-random==1.2.0 # 版本不对 mkl-service==2.3.0 # 找不到 sklearn==0.0 # 报错
> 
> absl-py 2.1.0 aiohttp 3.9.3 aiosignal 1.3.1 astunparse 1.6.3 async-timeout 4.0.3 attrs 23.2.0 blessed 1.20.0 blinker 1.7.0 boto3 1.34.72 botocore 1.34.72 Brotli 1.1.0 cached-property 1.5.2 cachetools 5.3.3 certifi 2024.2.2 cffi 1.16.0 charset-normalizer 3.3.2 click 8.1.7 contourpy 1.1.1 cryptography 42.0.5 cycler 0.12.1 easydict 1.12 filelock 3.13.1 flatbuffers 23.5.26 fonttools 4.49.0 frozenlist 1.4.1 fsspec 2024.2.0 gast 0.4.0 google-auth 2.28.1 google-auth-oauthlib 0.4.6 google-pasta 0.2.0 gpustat 1.1.1 grpcio 1.54.3 h5py 3.10.0 huggingface-hub 0.21.2 idna 3.6 importlib-metadata 7.0.1 importlib_resources 6.1.2 Jinja2 3.1.3 jmespath 1.0.1 joblib 1.3.2 keras 2.11.0 kiwisolver 1.4.5 libclang 16.0.6 Markdown 3.5.2 MarkupSafe 2.1.5 matplotlib 3.7.5 mpmath 1.3.0 multidict 6.0.5 networkx 3.1 nltk 3.8.1 numpy 1.24.4 nvidia-cublas-cu12 12.1.3.1 nvidia-cuda-cupti-cu12 12.1.105 nvidia-cuda-nvrtc-cu12 12.1.105 nvidia-cuda-runtime-cu12 12.1.105 nvidia-cudnn-cu12 8.9.2.26 nvidia-cufft-cu12 11.0.2.54 nvidia-curand-cu12 10.3.2.106 nvidia-cusolver-cu12 11.4.5.107 nvidia-cusparse-cu12 12.1.0.106 nvidia-ml-py 12.535.133 nvidia-nccl-cu12 2.19.3 nvidia-nvjitlink-cu12 12.3.101 nvidia-nvtx-cu12 12.1.105 oauthlib 3.2.2 opt-einsum 3.3.0 packaging 23.2 pandas 2.0.3 pillow 10.2.0 pip 24.0 protobuf 3.19.6 psutil 5.9.8 pyasn1 0.5.1 pyasn1-modules 0.3.0 pycparser 2.21 PyJWT 2.8.0 pyOpenSSL 24.0.0 pyparsing 3.1.1 PySocks 1.7.1 python-dateutil 2.8.2 pytorch-pretrained-bert 0.6.2 pytz 2024.1 pyu2f 0.1.5 PyYAML 6.0.1 regex 2023.12.25 requests 2.31.0 requests-oauthlib 1.3.1 rsa 4.9 s3transfer 0.10.1 safetensors 0.4.2 scikit-learn 1.3.2 scipy 1.10.1 sentence-transformers 2.4.0 setuptools 69.1.1 six 1.16.0 sympy 1.12 tensorboard 2.11.2 tensorboard-data-server 0.6.1 tensorboard-plugin-wit 1.8.1 tensorflow 2.13.1 tensorflow-estimator 2.11.0 tensorflow-io-gcs-filesystem 0.34.0 termcolor 2.4.0 threadpoolctl 3.3.0 tokenizers 0.15.2 torch 2.2.1 tqdm 4.66.2 transformers 4.38.1 triton 2.2.0 typing_extensions 4.10.0 tzdata 2024.1 urllib3 1.26.18 wcwidth 0.2.13 Werkzeug 3.0.1 wheel 0.42.0 wrapt 1.16.0 yarl 1.9.4 zipp 3.17.0 参考下这些包

谢谢，解决了，我这边环境为：
（1）环境为 python 3.8   pip 24.0
（2）2个依赖包得更新版本
mkl-random==1.2.2
mkl-service==2.4.0
（3）运行环境得能访问hunggingface，其中open_intent_detection/backbones/base.py  第38行，自动会去hugging face下载  bert-base-uncased 模型了，虽然我提前把模型离线下载到本地，但貌似没用，还好这模型小，不用等太久。
